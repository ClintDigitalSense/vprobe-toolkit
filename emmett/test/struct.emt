typedef struct teststruct {
   uint64 my64;
   uint8  my8;
} teststruct;

teststruct *foo;
teststruct **bar;

struct {
   uint32 b@4;
   uint32 c@12 - sizeof(uint32);
} *offptr;

VMM1Hz printf("barf\n", foo->my8, (*foo).my8, (*bar)->my64,
              &foo->my8, &(*bar)->my64);
VMMLoad printf("snarf: %d %d\n", &offptr->b, &offptr->c);
