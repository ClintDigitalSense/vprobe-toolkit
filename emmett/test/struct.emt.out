(version 0.2)
(definteger bar)
(definteger foo)
(definteger offptr)

 ;;;;;
(vprobe VMM1Hz ( printf    "barf\n"  (& 255 (getguest  (+ foo  8 )))  (& 255 (getguest  (+ foo  8 )))  (getguest (getguest bar))   (+ foo  8 )  (getguest bar) ))
 ;;;;;
(vprobe VMMLoad ( printf    "snarf: %d %d\n"   (+ offptr  4 )   (+ offptr  8 ) ))